zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/ZooKeeperEndpointGroupBuilder.java

```diff
@@ -15,58 +15,41 @@
  */
 package com.linecorp.armeria.client.zookeeper;
 
-import static com.google.common.base.Preconditions.checkState;
-import static com.linecorp.armeria.internal.common.zookeeper.ZooKeeperDefaults.DEFAULT_CONNECT_TIMEOUT_MS;
-import static com.linecorp.armeria.internal.common.zookeeper.ZooKeeperDefaults.DEFAULT_RETRY_POLICY;
-import static com.linecorp.armeria.internal.common.zookeeper.ZooKeeperDefaults.DEFAULT_SESSION_TIMEOUT_MS;
 import static java.util.Objects.requireNonNull;
 
-import java.util.ArrayList;
-import java.util.List;
+import java.time.Duration;
 import java.util.function.Consumer;
 
 import javax.annotation.Nullable;
 
 import org.apache.curator.framework.CuratorFramework;
-import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.framework.CuratorFrameworkFactory.Builder;
 
 import com.linecorp.armeria.client.endpoint.EndpointSelectionStrategy;
+import com.linecorp.armeria.common.zookeeper.AbstractCuratorFrameworkBuilder;
 import com.linecorp.armeria.common.zookeeper.NodeValueCodec;
 
 /**
  * Builds a {@link ZooKeeperEndpointGroup}.
  */
-public final class ZooKeeperEndpointGroupBuilder {
+public final class ZooKeeperEndpointGroupBuilder extends AbstractCuratorFrameworkBuilder {
 
     @Nullable
     private final CuratorFramework client;
```

Comment:
Perhaps we could move this field up and make `buildCuratorFramework()` return it? We could add a method like `isUserSpecifiedCuratorFramework()` to `AbstractCuratorFrameworkBuilder`.

Reply:
That sounds good.